pygmt/datatypes/header.py

```diff
@@ -0,0 +1,278 @@
+        # describing array layout in mem and interleaving
+        ("mem_layout", ctp.c_char * 4),
+        # Missing value as stored in grid file
+        ("nan_value", gmt_grdfloat),
+        # 0.0 for gridline grids and 0.5 for pixel grids
+        ("xy_off", ctp.c_double),
+        # Referencing system string in PROJ.4 format
+        ("ProjRefPROJ4", ctp.c_char_p),
+        # Referencing system string in WKT format
+        ("ProjRefWKT", ctp.c_char_p),
+        # Referencing system EPSG code
+        ("ProjRefEPSG", ctp.c_int),
+        # Lower-level information for GMT use only
+        ("hidden", ctp.c_void_p),
+    ]
+
+    def _parse_dimensions(self):
+        """
+        Get dimension names and attributes from the grid header.
+
+        For a 2-D grid, the dimension names are set to "y" and "x" by default. The
+        attributes for each dimension are parsed from the grid header following GMT
+        source codes. See the GMT functions "gmtnc_put_units", "gmtnc_get_units" and
+        "gmtnc_grd_info" for reference.
+        """
+        # Default dimension names.
+        dims = ("y", "x")
+        nameunits = (self.y_units, self.x_units)
+
+        # Dictionary for dimension attributes with the dimension name as the key.
+        attrs = {dim: {} for dim in dims}
+        # Dictionary for mapping the default dimension names to the actual names.
+        newdims = {dim: dim for dim in dims}
+        # Loop over dimensions and get the dimension name and attributes from header
+        for dim, nameunit in zip(dims, nameunits, strict=False):
+            # The long_name and units attributes.
+            long_name, units = _parse_nameunits(nameunit.decode())
+            if long_name:
+                attrs[dim]["long_name"] = long_name
+            if units:
+                attrs[dim]["units"] = units
+
+            # "degrees_east"/"degrees_north" are the units for geographic coordinates
+            # following CF-conventions
+            if units == "degrees_east":
+                attrs[dim]["standard_name"] = "longitude"
+                newdims[dim] = "lon"
+            elif units == "degrees_north":
+                attrs[dim]["standard_name"] = "latitude"
+                newdims[dim] = "lat"
+
+            # Axis attributes are "X"/"Y"/"Z"/"T" for horizontal/vertical/time axis.
+            attrs[dim]["axis"] = dim.upper()
+            idx = 2 if dim == "y" else 0
+            attrs[dim]["actual_range"] = np.array(self.wesn[idx : idx + 2])
+
+        # Save the lists of dimension names and attributes in the _nc attribute.
+        self._nc = {
+            "dims": [newdims[dim] for dim in dims],
+            "attrs": [attrs[dim] for dim in dims],
+        }
+
+    def get_name(self) -> str:
+        """
+        Get the name of the grid from the grid header.
+
+        Returns
+        -------
+        name
+            The name of the grid.
+        """
+        return "z"
```

Comment:
Can probably remove this if it is only returning the letter `z`?

Reply:
The name defaults to `z` for 2-D grids and `cube` for 3-D cubes, but it can be overridden by the `varname` variable stored in the hidden `GMT_GRID_HEADER_HIDDEN` data structure. I guess we won't wrap the `GMT_GRID_HEADER_HIDDEN` data structure now because it is designed to be *hidden* and may change anytime, but maybe we have to wrap it in the future?

Reply:
Right, I see that `varname` is marked as private at https://github.com/GenericMappingTools/gmt/blob/9b93e07982233a31e97b3f8a4ed8df3a3dfa56f9/src/gmt_hidden.h#L161, but it is supposedly holding the name of a NetCDF variable? Maybe we should try to parse it instead of always returning `z` (or `cube`)?

Reply:
As I understand it, `varname` is only assigned when a netCDF name like `input.nc?varname` is specified. I guess it's just rare cases.

As mentioned above, to parse `varname`, we have to wrap the `GMT_GRID_HEADER_HIDDEN` data structure, which contains more than 50 fields, and the last time it was changed is just a few months ago (https://github.com/GenericMappingTools/gmt/commit/b9d3f2bdc272e5f0c08dc93f45c622285fad9604).

Reply:
Got it. Let's just leave it unwrapped for now then.